### DISCUSSION TITLE
program has been unhealthy for 36s: restarting VPN

### DISCUSSION THREAD

**Discussion by torbengb** [qdm12/gluetun#1304] [Q&A] [answered]
- Running on a Synology NAS. Several other containers run fine.
- I am using "VPN Unlimited" and I am following [these steps from gluetun](https://github.com/qdm12/gluetun/wiki/VPN-Unlimited).  
```
docker run -it --rm --cap-add=NET_ADMIN -e VPN_SERVICE_PROVIDER="vpn unlimited" \
-v /yourpath/gluetun:/gluetun \
-e OPENVPN_USER=abc -e OPENVPN_PASSWORD=abc \
-e SERVER_COUNTRIES=Netherlands qmcgaw/gluetun
```
- I did of course provide my VPN account username and password into the command above.
- Program loops with these lines:

```
INFO [vpn] starting
INFO [firewall] allowing VPN connection...
INFO [openvpn] --cipher is not set. Previous OpenVPN version defaulted to BF-CBC as fallback when cipher negotiation failed in this case. If you need this fallback please add '--data-ciphers-fallback BF-CBC' to your configuration and/or add BF-CBC to --data-ciphers.
INFO [openvpn] OpenVPN 2.5.6 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 17 2022
INFO [openvpn] library versions: OpenSSL 1.1.1s  1 Nov 2022, LZO 2.10
INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]190.2.132.16:1194
INFO [openvpn] UDP link local: (not bound)
INFO [openvpn] UDP link remote: [AF_INET]190.2.132.16:1194
INFO [healthcheck] program has been unhealthy for 36s: restarting VPN
INFO [vpn] stopping
```

**Question:**

- What does "program is unhealthy" mean? How can I troubleshoot this?

**Comment by torbengb** [comment, marked as answer]
I think I found the problem: 

The certificates were created for France while the Docker command says `SERVER_COUNTRIES=Netherlands`. I changed the command to match the certificate, and then I saw `INFO [healthcheck] healthy!` so it appears to be working now. 

Case closed!

**Reply by qdm12** [reply]
So there is no bug/mismatch in the gluetun code right?

Regarding the healthy meaning, see https://github.com/qdm12/gluetun/wiki/Healthcheck
